packages/protocol/contracts/ACE/validators/joinSplitFluid/JoinSplitFluidABIEncoder.sol

```diff
@@ -156,7 +155,9 @@ library JoinSplitFluidABIEncoder {
             // 0x400 + metadataLength = byte length of output notes (0x120)
             // 0x420 + metadataLength = # of output notes (1)
             // 0x440 + metadataLength = offset to outputNotes[0] (0x60)
-            mstore(add(0x400, metadataLength), 0x120) // store length of output notes
+            let metadataLengthTemp := calldataload(add(metadataIndex, sub(metadata, 0x40)))
+
+            mstore(add(0x400, metadataLength), add(0x120, metadataLengthTemp)) // store length of output notes
```

Comment:
All these adds are superfluous no?

Reply:
yeah good catch, this was a short term debugging fix